plaso/parsers/text_plugins/snort_fastlog.py

```diff
@@ -0,0 +1,303 @@
+  """Snort3/Suricata fast-log alert event data.
+
+  Attributes:
+    classification (str): classification of the alert.
+    destination_ip (str): destination IP-address.
+    destination_port (int): destination TCP/UDP port number.
+    message (str): message associated with the alert.
+    priority (int): priorty, ranging from 1 (high) to 4 (very low).
+    rule_identifier (str): identifier of the Snort3/Suricata rule that generated
+        the alert.
+    source_ip (str): source IP-address.
+    source_port (int): source TCP/UDP port number.
+  """
+
+  DATA_TYPE = 'snort:fastlog:alert'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(SnortFastAlertEventData, self).__init__(data_type=self.DATA_TYPE)
+    self.classification = None
+    self.destination_ip = None
+    self.destination_port = None
+    self.message = None
+    self.priority = None
+    self.protocol = None
+    self.rule_identifier = None
+    self.source_ip = None
+    self.source_port = None
+
+
+class SnortFastLogTextPlugin(interface.TextPlugin):
+  """Text parser plugin for Snort3/Suricata fast-log alert log files."""
+
+  NAME = 'snort:fastlog:alert'
+  DATA_FORMAT = 'Snort3/Suricata fast-log alert log (fast.log) file'
+
+  _VERIFICATION_REGEX = re.compile(
+      # Date regex
+      r'^(\d{2}\/)?\d{2}\/\d{2}\-\d{2}:\d{2}:\d{2}.\d{6}\s*'
+      # Separator ([**])
+      r'\[\*\*\]\s*'
+      # Rule identifier
+      r'\[\d*:\d*:\d*\]\s*'
+      # Message
+      r'"?.*\"?\s*\[\*\*\]\s*'
+      # Optional Classification
+      r'(\[Classification:\s.*\])?\s*'
+      # Optional Priority
+      r'(\[Priority\:\s*\d{1}\])?\s*'
+      # Procotol
+      r'\{\w+\}\s*'
+      # Source IPv6 address
+      r'((([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:'
+      r'|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a -fA-F]{1,4}:){1,5}'
+      r'(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[ 0-9a-fA-F]{1,4})'
+      r'{1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4} |([0-9a-fA-F]'
+      r'{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((: [0-9a-fA-F]'
+      r'{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4})'
+      r'{0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2 [0-4]'
+      r'|1{0,1}[0-9]){0,1}[0-9]).){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}'
+      r'[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}'
+      r'[0-9]).){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))'
+      # Source IPv4 address
+      r'|(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?'
+      r'[0-9][0-9]?)\.(25[0-5] |2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|'
+      r'2[0-4][0-9]|[01]?[0-9][0-9]?))'
+      # Optional source port
+      r'(:\d*)?\s*'
+      # Separator '->'
+      r'\-\>\s*'
+      # Destination IPv6-address
+      r'((([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:'
+      r'|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a -fA-F]{1,4}:){1,5}'
```

Comment:
I assume this should be `0-9a-fA-F` not `0-9a -fA-F`

Reply:
Yes, absolutely. Thanks for correction in 7881955496450b9d78afbede9b2740247ec55766.